io/subsystem/src/main/java/org/wildfly/extension/io/WorkerAdd.java

```diff
@@ -189,7 +221,14 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
             }
         }
 
-        final WorkerService workerService = new WorkerService(builder.getMap());
+        OptionMap options = builder.getMap();
+        final ThreadsService threadsService = new ThreadsService(options.get(Options.WORKER_TASK_MAX_THREADS));
```

Comment:
Yes. One example I have in mind is to provide the EJB subsystem with the info so that strict max pools can auto-configure to match the sizing computed by the io worker pool.